amgut/static/js/interactive.js

```diff
@@ -0,0 +1,292 @@
+$(document).ready(function() {
+  //Patching in Mth.log2
+  Math.log2 = Math.log2 || function(x) {
+                return Math.log(x) / Math.LN2;
+            };
+  //Patching in findIndex for Internet Explorer
+  if (!Array.prototype.findIndex) {
+    Array.prototype.findIndex = function(predicate) {
+      if (this === null) {
+        throw new TypeError('Array.prototype.findIndex called on null or undefined');
+      }
+      if (typeof predicate !== 'function') {
+        throw new TypeError('predicate must be a function');
+      }
+      var list = Object(this);
+      var length = list.length >>> 0;
+      var thisArg = arguments[1];
+      var value;
+
+      for (var i = 0; i < length; i++) {
+        value = list[i];
+        if (predicate.call(thisArg, value, i, list)) {
+          return i;
+        }
+      }
+      return -1;
+    };
+  }
+});
+
+function group(list, prop) {
+  var grouped = {};
+  for(var i=0;i<list.length;i++) {
+    var key = list[i][prop];
+    grouped[key] = grouped[key] || [];
+    grouped[key].push(list[i]);
+  }
+  return grouped;
+}
+
+function getAvg(list) {
+  return list.reduce(function (x, y) {
+    return x + y;
+  }) / list.length;
+}
+
+function sumArrays(ar1, ar2) {
+  var sum = [];
+  if (ar1.length == ar2.length) {
+      for (var i=0; i<ar1.length;i++) {
+          sum.push(ar1[i] + ar2[i]);
+      }
+  }
+  else {
+    throw "Arrays unequal length: " + ar1.length + "  " + ar2.length;
+  }
+  return sum;
+}
+
+function findOtu(list, otu, level) {
+  var lvl = level || 'label';
+  function inner(element, index, array) {
+    if(element[lvl] == this) { return true; }
+    return false;
+  }
+  return list.findIndex(inner, otu);
+}
+
+function filterSites(list, sites) {
+
+}
+
+function collapse(dataset, level, max, prev_level, focus, sites) {
+  var phylum_colors = {
+    'Firmicutes': ['pink', 'darkred'],
+    'Bacteroidetes': ['#FFD27F', '#BE5E00'],
+    'Proteobacteria': ['lightyellow', 'yellow'],
+    'Actinobacteria': ['#F3DBAB', '#735B2B'],
+    'Verrucomicrobia': ['#C5D991', '#465A12'],
+    'Tenericutes': ['lightgreen', 'darkgreen'],
+    'Cyanobacteria': ['lightcyan', 'darkcyan'],
+    'Fusobacteria': ['lightblue', 'darkblue']
+  };
+  var phylum_order = ['Firmicutes', 'Bacteroidetes', 'Proteobacteria', 'Actinobacteria', 'Verrucomicrobia', 'Tenericutes', 'Cyanobacteria', 'Fusobacteria'];
+
+  var max_otus = max || 8;
+  var focus_taxa = focus || null;
+  var filter_to = sites || null;
+  var groups = group(dataset, level);
+  var collapsed = {};
+  var summaries = [];
+  for(var g in groups) {
+    if(focus !== null && groups[g][0][prev_level] != focus_taxa) { continue; }
+    var otu = { label: g,  data: groups[g][0].data, phylum: groups[g][0].phylum};
+    if(g === "") { otu.label = 'Unspecified'; }
+    // Collapse the OTUs found in the group into a single
+    for(var i=1;i<groups[g].length;i++) {
```

Comment:
I'm not sure what style guidelines are in place for the javascript code being developed, but at a minimum, would it be possible to be consistent on how statements are done? For instance, there are instances of both of the following (as well as for if statements):

```javascript
for(...) { /* do stuff */ };
for(...) {
    /* do stuff */
};
```